engine/src/main/java/org/terasology/engine/world/block/BlockArea.java

```diff
@@ -116,6 +116,7 @@ public Vector2ic next() {
 
                 if (current.equals(next)) {
                     if (findNext()) {
+                        current.set(next);
```

Comment:
can you add a test for this case. I think this doesn't cover the case if you call next() twice in a row without findNext(). I think I made the same mistake in BlockRegion.